powerauth-push-server/src/main/java/io/getlime/push/service/EnhancedPushConfiguration.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright 2016 Lime - HighTech Solutions s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getlime.push.service;
+
+import com.turo.pushy.apns.ApnsClient;
+import com.turo.pushy.apns.ApnsClientBuilder;
+import com.turo.pushy.apns.auth.ApnsSigningKey;
+import com.turo.pushy.apns.proxy.HttpProxyHandlerFactory;
+import io.getlime.push.configuration.PushServiceConfiguration;
+import io.getlime.push.errorhandling.exceptions.PushServerException;
+import io.getlime.push.model.entity.PushMessageAttributes;
+import io.getlime.push.model.entity.PushMessageBody;
+import io.getlime.push.service.fcm.FcmClient;
+import io.getlime.push.service.fcm.FcmNotification;
+import io.getlime.push.service.fcm.model.FcmSendRequest;
+import io.getlime.push.service.fcm.model.FcmSendResponse;
+import io.getlime.push.service.fcm.model.base.FcmResult;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+
+import javax.net.ssl.SSLException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class EnhancedPushConfiguration {
```

Comment:
I like where this change is going but I'm a bit unsure about the class naming. This class does push message sending and as such, it is rather `*Worker` than `*Configuration`. Let's call the class `PushSendingWorker` instead.